src/OpenSage.Game/Terrain/Roads/TextureAtlas.cs

```diff
@@ -0,0 +1,155 @@
+﻿using System.Collections.Generic;
+using System.Numerics;
+using OpenSage.Mathematics;
+
+namespace OpenSage.Terrain.Roads
+{
+    internal class TextureAtlas
+    {
+        // The actual texture coordinates depend on the RoadTemplate's RoadWidthInTexture,
+        // so we create and cache one instance per width.
+        private static readonly IDictionary<float, TextureAtlas> TextureAtlasCache = new Dictionary<float, TextureAtlas>();
+
+        // The default road width in texture space.
+        private const float UnscaledRoadWidth = 0.25f;
+
+        // The length of the incoming roads in T and X crossings in texture space.
+        private const float IncomingRoadStubLength = 0.004f;
+
+        // The length from the center to the endpoint of the incoming roads in T and X crossings in texture space. 
+        private const float LengthToIncomingRoad = UnscaledRoadWidth / 2 + IncomingRoadStubLength;
+
+        // There are 3x3 tiles in a road texture, so a half tile is 1/6 = 0.1666...,
+        // but the calculations are based on the rounded number.
+        private const float HalfTileSize = 0.166f;
```

Comment:
Shouldn't the rounded number be 0.167f?

Reply:
Yeah, it should :D It isn't though...for whatever reason. I used the unrounded 1/6 for a long time and never know why the coordinates were a little off.